pkg/util/deployment/deployment.go

```diff
@@ -140,11 +140,31 @@ func ListPods(deployment *extensions.Deployment, getPodList podListFunc) (*api.P
 	return getPodList(namespace, options)
 }
 
+// equalIgnoreHash returns true if two given podTemplateSpec are equal, ignoring the diff in value of Labels[pod-template-hash]
+// We ignore pod-template-hash because the hash result would be different upon podTemplateSpec API changes
+// Note that we assume input podTemplateSpecs contain non-empty labels
+func equalIgnoreHash(template1, template2 api.PodTemplateSpec) (bool, error) {
+	if len(template1.Labels) == 0 || len(template2.Labels) == 0 {
```

Comment:
Add a comment about why these will never be empty.  (The PT must have a label so label selectors can find them.  This is checked by validation.)